testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java

```diff
@@ -2749,7 +2749,7 @@ public void testUpdateRowConcurrently()
                                 verifyConcurrentUpdateFailurePermissible(trinoException);
                             }
                             catch (Throwable verifyFailure) {
-                                if (trinoException != e && verifyFailure != e) {
```

Comment:
Change looks good to me.

Only thing coming in my mind is that trinoException might be equal to e but that implies that the exception has a cycle/self-reference?

e.g.

```java
catch (Exception e) {
  RuntimeException trinoException = getTrinoExceptionCause(e);
  e.addSuppressed(trinoException);
  throw new TrinoException(..., e);
}
```

Reply:
it's possible & legit for `e == trinoException`:
```
RuntimeException trinoException = getTrinoExceptionCause(e);
```

simply `e` may be `TrinoException` itself.